### PR TITLE
api: vod asset Delete API & event

### PR DESCRIPTION
**Pull request by gioelecerati**
**What does this pull request do? Explain your changes. (required)**

Spawn delete event for an Asset on api call

**Specific updates (required)**

- Delete event
- All children of deleted asset become orphans

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional):**

**Checklist:**

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/schema/schema.yaml [hunk 1]
```diff
@@ -1140,6 +1141,15 @@ components:
               properties:
                 profile:
                   $ref: "#/components/schemas/ffmpeg-profile"
+            delete:
```
Comment (gioelecerati) [inline packages/api/src/schema/schema.yaml:1144]: Should the delete task also have an output object?

File: packages/api/src/controllers/asset.ts [hunk 2]
```diff
@@ -471,9 +471,36 @@ app.delete("/:id", authMiddleware({}), async (req, res) => {
...
+  );
+
+  if (!childrenAssets?.length || !childrenAssets[0]?.length) {
+    res.status(200);
+    return res.json({ task });
+  }
+
+  for (var i = 0; i < childrenAssets[0].length; i++) {
+    // remove the sourceAssetId from the children assets making them orphans
+    db.asset.update(childrenAssets[0][i].id, {
```
Comment (gioelecerati) [inline packages/api/src/controllers/asset.ts:497]: Should be made in a single query

File: packages/api/src/controllers/asset.ts [hunk 3]
```diff
@@ -471,9 +471,36 @@ app.delete("/:id", authMiddleware({}), async (req, res) => {
...
+    },
+    asset
+  );
+
+  if (!childrenAssets?.length || !childrenAssets[0]?.length) {
+    res.status(200);
+    return res.json({ task });
+  }
+
+  for (var i = 0; i < childrenAssets[0].length; i++) {
```
Comment (victorges) [inline packages/api/src/controllers/asset.ts:495]: **nit:**
```suggestion
  for (var i in childrenAssets[0]) {
```
to keep `i` as the index, or
```suggestion
  for (var child of childrenAssets[0]) {
```
to already get the asset in the `child` var

File: packages/api/src/controllers/asset.ts (hunk 2 above)
Comment (victorges) [reply packages/api/src/controllers/asset.ts:497]: Yeah that's one pretty doable actually. We could first get the id of the children, then update all the children + the source on a single query and avoid any dangling errors. Might not be super trivial to implement it right now tho, so maybe just a `// TODO` here should be ok

File: packages/api/src/schema/schema.yaml (hunk 1 above)
Comment (victorges) [reply packages/api/src/schema/schema.yaml:1144]: Hmm I don't think so

**Review by victorges — APPROVED** [review]
lgtm

**Comment by pglowacky** [comment]
@gioelecerati what's the status of this?

**Pull request closed without merging**